lms/djangoapps/courseware/access_utils.py

```diff
@@ -63,8 +63,11 @@ def check_start_date(user, days_early_for_beta, start, course_key):
         return ACCESS_GRANTED
     else:
         now = datetime.now(UTC())
+        if start is None or in_preview_mode():
+            return ACCESS_GRANTED
+
         effective_start = adjust_start_date(user, days_early_for_beta, start, course_key)
-        if start is None or now > effective_start or in_preview_mode():
+        if now > effective_start:
```

Comment:
@mushtaqak I just split the 3 OR conditions.  `adjust_start_date` will fail if `start` is `None`. As the check existed before that if start is None then grant access. 
This means `adjust_start_date` will not execute and break if start is found to be None.